server/server.go

```diff
@@ -458,8 +470,16 @@ func (s *Server) setClusterName(name string) {
 	}
 	s.info.Cluster = name
 	s.routeInfo.Cluster = name
+
+	// Marks whether this is a dynamic name change that is
+	// susceptible to changes due to INFO/CONNECT messages
+	// from other nodes in the cluster.
+	s.susceptible = susceptible
+	s.routeInfo.Dynamic = susceptible
+
```

Comment:
name change suggested by Ivan to apply to `s.susceptible` as well as `susceptible`